azure-kusto-data/source/client.ts

```diff
@@ -18,23 +18,25 @@ const MGMT_PREFIX = ".";
 enum ExecutionType {
     Mgmt = 0,
     Query = 1,
-    Ingest = 2
+    Ingest = 2,
+    QueryV1 = 3,
 }
 
 export class KustoClient {
     connectionString: ConnectionStringBuilder;
     cluster: string;
-    endpoints: { mgmt: string; query: string; ingest: string; };
+    endpoints: { [key in ExecutionType] : string; };
```

Comment:
I like using cool feautres of typescript, but note that because we use them as keys in an object, they are stringified.
So `endpoints` has the keys "0", and "1" and so on, and I don't really like it.
Making it a string enum might be nicer, but it's your choice.
```ts
enum ExecutionType {
    Mgmt = "mgmt",
    Query = "query",
    Ingest = "ingest",
    QueryV1 = "queryv1",
}
```